backends/ubpf/p4c-ubpf.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2019 Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <stdio.h>
+#include <string>
+#include <iostream>
+
+#include "backends/ubpf/version.h"
+#include "ir/ir.h"
+#include "lib/log.h"
+#include "lib/gc.h"
+#include "lib/crash.h"
+#include "lib/exceptions.h"
+#include "lib/nullstream.h"
+
+#include "backends/ebpf/midend.h"
+#include "backends/ebpf/ebpfOptions.h"
+#include "ubpfBackend.h"
+#include "frontends/p4/frontend.h"
+#include "frontends/common/applyOptionsPragmas.h"
+#include "frontends/common/parseInput.h"
+#include "ir/json_loader.h"
+#include "fstream"
+
+void compile(EbpfOptions& options) {
+    auto hook = options.getDebugHook();
+    bool isv1 = options.langVersion == CompilerOptions::FrontendVersion::P4_14;
+    if (isv1) {
+        ::error("This compiler only handles P4-16");
```

Comment:
you should consider migrating to the function that takes an `ErrorType::ERR_*` argument, but there is no hurry.

Reply:
Done.